tests/v2/validation/scaling/scaling_node_driver_test.go

```diff
@@ -0,0 +1,100 @@
+package scaling
+
+import (
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters/kubernetesversions"
+	"github.com/rancher/rancher/tests/framework/extensions/provisioninginput"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	password "github.com/rancher/rancher/tests/framework/extensions/users/passwordgenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type ScalingTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	standardUserClient *rancher.Client
+	session            *session.Session
+	clustersConfig     *provisioninginput.Config
+}
+
+func (s *ScalingTestSuite) TearDownSuite() {
+	s.session.Cleanup()
+}
+
+func (s *ScalingTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	s.session = testSession
+
+	s.clustersConfig = new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, s.clustersConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(s.T(), err)
+
+	s.client = client
+
+	s.clustersConfig.K3SKubernetesVersions, err = kubernetesversions.Default(s.client, clusters.K3SClusterType.String(), s.clustersConfig.K3SKubernetesVersions)
+	require.NoError(s.T(), err)
+
+	s.clustersConfig.RKE1KubernetesVersions, err = kubernetesversions.Default(s.client, clusters.RKE1ClusterType.String(), s.clustersConfig.RKE1KubernetesVersions)
+	require.NoError(s.T(), err)
+
+	s.clustersConfig.RKE2KubernetesVersions, err = kubernetesversions.Default(s.client, clusters.RKE2ClusterType.String(), s.clustersConfig.RKE2KubernetesVersions)
+	require.NoError(s.T(), err)
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(s.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(s.T(), err)
+
+	s.standardUserClient = standardUserClient
+}
+
+func (s *ScalingTestSuite) TestScalingRKE2K3SWorkerNodePools() {
+	if len(s.clustersConfig.NodesAndRoles) == 0 {
+		s.T().Skip()
+	}
```

Comment:
from what I saw, the nodesAndRoles were hard-coded in the test. Is this necessary or did I misunderstand? Confusing since this uses an existing cluster.

Reply:
This can be probably be removed for that reason. I'll test that out and remove if that's the case.